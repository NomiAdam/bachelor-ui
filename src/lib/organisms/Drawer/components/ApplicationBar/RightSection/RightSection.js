import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Label from './Label/index';
import Button from '../../../../../atoms/Button/index';

export const StyledRightSection= styled.div`
	width: 50%;
	margin-right: 10px;
	display: flex;
	flex-direction: row;
	flex-wrap: nowrap;
	justify-content: flex-end;
	align-items: center;
`;

const RightSection = ({ onLogout, userName}) => (
    <StyledRightSection>
        <Label
            username={userName}
        />
        <Button
            label="odhlÃ¡sit se"
            onClick={onLogout}
        />
    </StyledRightSection>
);

RightSection.propTypes = {
    /**
     * Displays name of logged user
     */
    userName: PropTypes.string,
    /**
     * Logout function
     */
    onLogout: PropTypes.func,
};

export default RightSection;
