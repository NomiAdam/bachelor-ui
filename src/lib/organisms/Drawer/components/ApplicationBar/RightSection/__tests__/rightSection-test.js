/* eslint-disable no-undef */
import React from 'react';
import RightSection from '../RightSection';
import { shallow } from 'enzyme';

describe('RightSection', () => {
    it('should be defined', () => {
        expect(RightSection).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<RightSection />)).toMatchSnapshot();
    });
});
