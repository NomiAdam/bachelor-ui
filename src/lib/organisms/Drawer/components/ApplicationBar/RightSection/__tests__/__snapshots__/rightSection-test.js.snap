// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSection should match the snapshot 1`] = `
<styled.div>
  <Label />
  <Button
    background="#0247fe"
    fullWidth={false}
    label="odhlÃ¡sit se"
  />
</styled.div>
`;
